Description: Look for vulkan validationlayers header to support Vulkan
 Avoid vulkan support if no validationlayers is found in the building env
Author: Jose Luis Rivero <jrivero@openrobotics.org>
Forwarded: not-needed
Last-Update: 2022-05-19
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/CMake/Packages/FindVulkan.cmake
+++ b/CMake/Packages/FindVulkan.cmake
@@ -101,6 +101,13 @@
 		)
 endif()
 
+# Skip vulkan support in the building enviroments wihtout validationlayers
+# In Ubuntu Focal and lower, validationlayers version in control file can not
+# satisfy the requirement and fallback to use a workaround. We need to skip
+# this support if the validationlayers headers are not found.
+find_path(VulkanLayers_INCLUDE_DIR
+  NAMES vulkan/vk_layer_config.h)
+
 set(Vulkan_LIBRARIES ${Vulkan_LIBRARY})
 set(Vulkan_INCLUDE_DIRS ${Vulkan_INCLUDE_DIR})
 
@@ -122,3 +129,10 @@
   add_executable(Vulkan::glslc IMPORTED)
   set_property(TARGET Vulkan::glslc PROPERTY IMPORTED_LOCATION "${Vulkan_GLSLC_EXECUTABLE}")
 endif()
+
+if (NOT VulkanLayers_INCLUDE_DIR)
+  message(STATUS "VulkanLayers not found. Disabling Vulkan support")
+  set(Vulkan_FOUND FALSE)
+endif()
+
+
